core/transaction_pool/impl/pool_moderator_impl.hpp

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_POOL_MODERATOR_IMPL_HPP
+#define KAGOME_POOL_MODERATOR_IMPL_HPP
+
+#include <map>
+
+#include "transaction_pool/impl/clock_impl.hpp"
+#include "transaction_pool/pool_moderator.hpp"
+
+namespace kagome::transaction_pool {
+
+  class PoolModeratorImpl : public PoolModerator {
+    static constexpr size_t kDefaultExpectedSize = 2048;
+
+    using SystemClock = Clock<std::chrono::system_clock>;
+    using Map = std::map<common::Hash256, SystemClock::TimePoint>;
+
+   public:
+    /**
+     * @param ban_for amount of time for which a transaction is banned
+     * @param clock a clock used to determine when it is time to unban a
+     * transaction
+     * @param expected_size expected maximum number of banned transactions. If
+     * significantly exceeded, some transactions will be removed from ban list
+     */
+    explicit PoolModeratorImpl(
+        SystemClock::Duration ban_for = std::chrono::minutes(30),
```

Comment:
maybe use literal?
like
```
using std::chrono_literals::operator""m;
30m

```

Reply:
While it's used only once in this file, I don't think that using literals is a much better option. The amount of code would only increase, and to me the way it is now seems quite readable.